scaladoc/src/dotty/tools/scaladoc/site/blocks/Tabs.scala

```diff
@@ -0,0 +1,89 @@
+package dotty.tools.scaladoc.site.blocks
+
+
+import liqp.TemplateContext
+import liqp.nodes.LNode
+import liqp.blocks.Block
+import scala.collection.mutable.ListBuffer
+import java.util.UUID
+
+class TabsBlock extends Block("tabs") {
+  override def render(context: TemplateContext, nodes: LNode*): AnyRef = {
+    // Generate a unique ID for this set of tabs
+    val uniqueId = UUID.randomUUID().toString
+
+    // Add the unique ID to the context
+    context.put("tabs-unique-id", uniqueId)
+
+    // Render the content of the tabs block
+    nodes.foreach(_.render(context))
+
+    // Extract tab headers and contents from the context
+    val tabHeaders = context.remove(s"tab-headers-$uniqueId").asInstanceOf[ListBuffer[String]].mkString("\n")
+    val tabContents = context.remove(s"tab-contents-$uniqueId").asInstanceOf[ListBuffer[String]].mkString("\n")
+
+    // Build the HTML string using StringBuilder
+    val builder = new StringBuilder
+    builder.append(
+      s"""
+         |<div class="tabs" id="tabs-$uniqueId">
+         |  $tabHeaders
+         |  $tabContents
+         |</div>
+         |""".stripMargin)
+
+    // Return the final rendered string
+    builder.toString
+  }
+}
+
+class TabBlock extends Block("tab") {
+  override def render(context: TemplateContext, nodes: LNode*): AnyRef = {
```

Comment:
there was some additional unported validation logic for acceptable combinations of tab labels, but that was pretty specialised to the needs of docs.scala-lang.org - so it would be hard to see how to make that configurable (perhaps by yaml?)

Reply:
something like a matcher (if one of these labels exists, then ensure that tabs are arranged in this order..)